src/workspaceViewer.ts

```diff
@@ -92,16 +93,22 @@ export class WorkspaceItem extends TreeItem {
 		dim?: number[]
 	) {
 		super(label, collapsibleState);
-		this.description = this.getDescription(dim, str);
+		this.description = this.getDescription(dim, str, rClass);
 		this.tooltip = `${label} (${rClass}, length of ${length})`;
 		this.contextValue = type;
 	}
 
-	private getDescription(dim: number[], str: string): string {
+	private getDescription(dim: number[], str: string, rClass: string): string {
 		if (dim !== undefined) {
```

Comment:
In RStudio, it looks like only data.frame will get a description of `n obs. m variables` and array will use str as its description. It makes sense because matrix and array may not fit the description in terms of nobs and variables but number of rows and columns, or other dimensions.

I suggest we adopt this way. Only `list` type with `dim` uses `class: n obs. m variables` description, otherwise use its `str` instead.